acceptance/tests/reports/corrective_change_via_puppet.rb

```diff
@@ -0,0 +1,98 @@
+
+  test_file_name = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, test_file_name)
+  tmp_file = {}
+  original_test_data = 'this is my original important data'
+  modified_test_data = 'this is my modified important data'
+  
+  agents.each do |agent|
+    tmp_file[agent.hostname] = agent.tmpfile(tmp_environment)
+  end
+
+  teardown do
+    step 'clean out produced resources' do
+      agents.each do |agent|
+       if tmp_file.has_key?(agent.hostname) && tmp_file[agent.hostname] != ''  
+        on(agent, "rm #{tmp_file[agent.hostname]}", :accept_all_exit_codes => true)
+      end 
+     end
+    end
+  end
+
+  def create_manifest_for_file_resource(file_resource, file_contents, environment_name)
+    manifest = <<MANIFEST
+file { '#{environmentpath}/#{environment_name}/manifests/site.pp':
+  ensure => file,
+  content => '
+\$test_path = \$::fqdn ? #{file_resource}
+file { \$test_path:
+  content => @(UTF8)
+    #{file_contents}
+    | UTF8
+}
+  ',
+}
+MANIFEST
+    apply_manifest_on(master, manifest, :catch_failures => true)
+  end
+
+  step 'create file resource in site.pp' do
+    create_manifest_for_file_resource(tmp_file, original_test_data, tmp_environment)
+  end
+
+  step 'run agent(s) to create the new resource' do
+    with_puppet_running_on(master, {}) do
+      agents.each do |agent|
+        fqdn = agent.hostname
+
+        step 'Run agent once to create new File resource' do
+          on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname}"),:acceptable_exit_codes => 2)
+        end
+
+        step 'Verify the file resource is created' do
+          on(agent, "cat #{tmp_file[fqdn]}").stdout do |file_contents|
+            assert_equal(original_test_data, file_contents, 'file contents did not match expected contents')
+          end
+        end
+
+        step 'Change the manifest for the resource' do
+          create_manifest_for_file_resource(tmp_file, modified_test_data, tmp_environment)
+        end
+
+        step 'Run agent a 2nd time to change the File resource' do
+          on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname}"),:acceptable_exit_codes => 2)
+        end
+
+        step 'Verify the file resource is created' do
+         on(agent, "cat #{tmp_file[fqdn]}").stdout do |file_contents|
+           assert_equal(modified_test_data, file_contents, 'file contents did not match expected contents')
+         end
+        end
+      end
+    end
+  end
+
+  # Open last_run_report.yaml
+  step 'Check report' do
+     agents.each do |agent|
+       on(agent, puppet('config print statedir')) do |command_result|
+         report_path = command_result.stdout.chomp + '/last_run_report.yaml'
+         on(agent, "cat #{report_path}") do |report_yaml|
+
+           #stripping out ruby class definitions from the report yaml
+           report_yaml = YAML.load(report_yaml.stdout.gsub(/\!ruby[^\n]*/, ''))
```

Comment:
Take a look at this: https://tickets.puppetlabs.com/browse/PCP-542?focusedCommentId=325494&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-325494 ... I'm implementing similar over here, but specifically for !ruby: https://github.com/puppetlabs/puppet/pull/5220/files#diff-a5348c3071a6b3d7b77bf0ed94147edfR15